src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -2349,13 +2349,23 @@ C2V_VMENTRY_PREFIX(jlong, getCurrentJavaThread, (JNIEnv* env, jobject c2vm))
   return (jlong) p2i(thread);
 C2V_END
 
-C2V_VMENTRY_PREFIX(jboolean, attachCurrentThread, (JNIEnv* env, jobject c2vm, jboolean as_daemon))
+C2V_VMENTRY_PREFIX(jboolean, attachCurrentThread, (JNIEnv* env, jobject c2vm, jbyteArray name, jboolean as_daemon))
   if (thread == NULL) {
     // Called from unattached JVMCI shared library thread
     extern struct JavaVM_ main_vm;
     JNIEnv* hotspotEnv;
-    jint res = as_daemon ? main_vm.AttachCurrentThreadAsDaemon((void**) &hotspotEnv, NULL) :
-                           main_vm.AttachCurrentThread((void**) &hotspotEnv, NULL);
+
+    int name_len = env->GetArrayLength(name);
```

Comment:
`name` could be NULL based on code in attachCurrentThread. Should we check for NULL here?

Reply:
It's not obvious, but the test against `IS_IN_NATIVE_IMAGE` in `HotSpotJVMCIRuntime.attachCurrentThread` is equivalent to the `thread == NULL` test here. I'll add a comment (and maybe a `guarantee`) to clarify this.